### ISSUE TITLE
PackageCompiler fails nightly PkgEval (but nightly CI passes in this repo's GitHub Actions)

### ISSUE THREAD

**Issue by DilumAluthge** [JuliaLang/PackageCompiler.jl#1020]
I don't know if this is a PkgEval bug or a PackageCompiler bug (or a Julia bug).

In PkgEval on `julia#master`, PackageCompiler times out, even if the timeout is really high. For example[, this PkgEval job](https://github.com/JuliaLang/julia/commit/6cb9f04c1ec562f800669ba13a9a096203aea7cf#commitcomment-150098037) times out at 8181 seconds:

> PkgEval terminated after 8181.26s: test duration exceeded the time limit

8181 seconds = 2 hours 16 minutes.

However, when I run PackageCompiler on GitHub Actions CI on Julia nightly, the job passes in 1 hour 27 minutes. See for example [this recent nightly CI job on this repo](https://github.com/JuliaLang/PackageCompiler.jl/actions/runs/12216485102/job/34079649170?pr=1017):

> succeeded 3 days ago in 1h 27m 30s

It is unfortunate that PackageCompiler fails in nightly PkgEval, because it means we can't detect if/when something in upstream Julia breaks PkgEval.

@maleadt @KristofferC Any ideas?

**Comment by maleadt** [comment]
Looking at the server now that the job has been running for close to 90 minutes, it doesn't seem stuck, just chugging along. Maybe it's slow because of enabling assertions in Julia? In any case, I don't want to extend the PkgEval time limits any further, even if it's only a single package exceeding them, the "tail latency" is large enough already (so testing PackageCompiler is likely to block the entire process while it's the only package being tested).

**Comment by DilumAluthge** [comment]
Could we use a regular (non-assert) build of Julia when testing PackageCompiler in PkgEval?

**Comment by maleadt** [comment]
No, that's the kind of hacking around I want to avoid adding to PkgEval.

**Comment by KristofferC** [comment]
We can make

https://github.com/JuliaLang/PackageCompiler.jl/blob/1c35331d8ef81494f054bbc71214811253101993/test/runtests.jl#L28

be true on PkgEval?

**Comment by KristofferC** [comment]
But that is not the only condition?

**Comment by maleadt** [comment]
Oh, sorry, I'm blind. Yeah, it's probably worth extending that to cover PkgEval.
